### PR TITLE
Booking view and App TypeScript

### PR DESCRIPTION
**Pull request by devmount**
## Description of the Change

This change adds types for all booking view components and `App.vue`.

- The script section was moved to the top in the vue files
- Typed keys were provided while keeping old keys for compatibility until all files are typed

## Applicable Issues

Implements #413

### CODE REVIEW

File: frontend/src/App.vue [hunk 1]
```diff
@@ -73,9 +44,10 @@ const isAuthenticated = computed(() => currentUser?.exists());
 const call = createFetch({
   baseUrl: apiUrl,
   options: {
-    async beforeFetch({ options }) {
+    beforeFetch({ options }) {
       if (isAuthenticated.value) {
-        const token = await currentUser.data.accessToken;
```
Comment (devmount) [inline frontend/src/App.vue:78]: This doesn't return a promise to wait for.

File: frontend/src/App.vue [hunk 2]
```diff
@@ -73,9 +44,10 @@ const isAuthenticated = computed(() => currentUser?.exists());
...
+        const token = currentUser.data.accessToken;
+        // @ts-ignore
         options.headers.Authorization = `Bearer ${token}`;
```
Comment (devmount) [inline frontend/src/App.vue:51]: Sorry, my first ts-ignore 🙈 Somehow `Authorization` isn't recognized here as valid header option and I didn't found a way to properly type that within reason.

File: frontend/src/App.vue [hunk 3]
```diff
@@ -104,7 +76,7 @@ const call = createFetch({
         );
       } else if (response && response.status === 401 && data?.detail?.id === 'INVALID_TOKEN') {
         // Clear current user data, and ship them to the login screen!
-        await currentUser.$reset();
```
Comment (devmount) [inline frontend/src/App.vue:107]: Again no promise here.

File: frontend/src/App.vue [hunk 4]
```diff
@@ -119,13 +91,19 @@ const call = createFetch({
...
 
+// TODO: Deprecated - Please use isPasswordAuthKey, as it's typed!
 provide('isPasswordAuth', import.meta.env?.VITE_AUTH_SCHEME === 'password');
+provide(isPasswordAuthKey, import.meta.env?.VITE_AUTH_SCHEME === 'password');
+// TODO: Deprecated - Please use isFxaAuthKey, as it's typed!
 provide('isFxaAuth', import.meta.env?.VITE_AUTH_SCHEME === 'fxa');
+provide(isFxaAuthKey, import.meta.env?.VITE_AUTH_SCHEME === 'fxa');
+// TODO: Deprecated - Please use fxaEditProfileUrlKey, as it's typed!
 provide('fxaEditProfileUrl', import.meta.env?.VITE_FXA_EDIT_PROFILE);
+provide(fxaEditProfileUrlKey, import.meta.env?.VITE_FXA_EDIT_PROFILE);
```
Comment (devmount) [inline frontend/src/App.vue:106]: I remove those todos as soon as the containing files are typed.

File: frontend/src/components/bookingView/BookingViewSlotSelection.vue [hunk 5]
```diff
@@ -1,3 +1,51 @@
...
+ */
+const selectEvent = (day: String) => {
+  // set event selected
+  for (let i = 0; i < appointment.value.slots.length; i += 1) {
+    const slot: Slot = appointment.value.slots[i];
+    if (slot.start.format(dateFormatStrings.qalendar) === day) {
+      slot.selected = true;
+      const e = { ...appointment.value, ...slot };
+      delete e.slots;
+      selectedEvent.value = e;
```
Comment (devmount) [inline frontend/src/components/bookingView/BookingViewSlotSelection.vue:40]: For now I typed that as `Appointment & Slot`, but we might want to evaluate if we even need an intersection type here.

File: frontend/src/components/bookingView/BookingViewSlotSelection.vue [hunk 6]
```diff
@@ -20,7 +68,7 @@
       :current-date="activeDate"
       :appointments="[appointment]"
       :is-booking-route="true"
-      :fixed-duration="appointment?.slot_duration ?? activeSchedules[0]?.slot_duration"
+      :fixed-duration="activeSchedules[0]?.slot_duration"
```
Comment (devmount) [inline frontend/src/components/bookingView/BookingViewSlotSelection.vue:71]: This was from one-off appointments I think. Removed it.

**Review by devmount — COMMENTED** [review]
Some notes on this.

**Review by MelissaAutumn — APPROVED** [review]
Looks great, thanks!

**Pull request merged**